### DISCUSSION TITLE
"Shortcuts" when navigating with a GPX

### DISCUSSION THREAD

**Discussion by barteis** [osmandapp/OsmAnd#11174] [Q&A]
When navigating with a GPX off-road, the route "shortcuts" the actual trails as illustrated below.  The problem occurs ahead of my current position and clears up as I pass through the region.  I've seen with more than on more than one GPX, but not always.  It doesn't seem to occur on "simplified tracks" (I think that's what they're called) that are saved without points.  But the points are useful for getting turn-by-turn directions.  Any ideas?  

![shortcut 1](https://user-images.githubusercontent.com/36173950/111352176-2794ee00-8641-11eb-84b4-f6f15a9e964b.png)
![shortcut 2](https://user-images.githubusercontent.com/36173950/111352184-2b287500-8641-11eb-9688-fa3581954fcc.png)

**Comment by vshcherb** [comment]
This would be great if it's reproducible with the track you've recorded i.e. simulate your position with GPX track will get this behavior. It looks wrong but the reason is unclear whether it's internal or external (jumping gps location).

**Reply by barteis** [reply]
I no longer have that GPX, but I'll do some testing.  This was a forested area, which could possibly affect GPS precision. I'll try some open areas, as well.

**Reply by vshcherb** [reply]
To reproduce the issue 2 GPX is needed: navigation & recorded, so we can replay the navigation

**Reply by barteis** [reply]
What if I plan a route with Trip planner to create a GPX. Then  navigate with it while recording a second GPX.  Would that work?

**Reply by vshcherb** [reply]
Nope, I think there is a legitimate issue we just need to reproduce it to answer all questions. I don't see it's related to origin of GPX.

**Reply by barteis** [reply]
Here's another example, together with the GPX track and route files. (Github doesn't seem to support GPX uploads, so these are via Google Drive.)  This uses the California standard map.

https://drive.google.com/drive/folders/1MozQaKAJ8CTX-qU6vUBihXYDzOq_bmg8?usp=sharing

![Shortcut route](https://user-images.githubusercontent.com/36173950/111885501-6ef2e580-8985-11eb-8c32-45148b413968.png)

**Reply by vshcherb** [reply]
Yes

**Reply by vshcherb** [reply]
https://github.com/osmandapp/OsmAnd/issues/11235
